src/pages/index.js

```diff
@@ -30,49 +32,126 @@ const profileDescriptionInput = document.querySelector(
+    userInfo.setUserImage(userData.avatar);
+  })
+  .catch(console.error);
+
+/*=============================================
+=           Create Cards          =
+=============================================*/
+//fetchnewcard
 function createCard(cardData) {
   const cardElement = new Card(
     cardData,
     "#card-template",
-    handlePreviewPicture
+    handlePreviewPicture,
+    handleDeleteClick,
+    handleLikeClick
   );
   return cardElement.getView();
 }
 
 /*=============================================
-=            User Info            =
+=           profile Info            =
 =============================================*/
 
-const userInfo = new UserInfo(".profile__title", ".profile__description");
-
-function handleProfileEditSubmit(UserData) {
-  profilePopupForm.close();
-  userInfo.setUserInfo(UserData);
+function handleProfileEditSubmit(userData) {
+  profilePopupForm.setLoading(true);
+  api
+    .fetchEditProfile(userData)
+    .then((userData) => {
+      profilePopupForm.close(userData);
+    })
+    .catch((err) => console.error(err))
+    .finally(() => profilePopupForm.setLoading(false));
 }
 
 /*=============================================
-=            section            =
+=            profile image edit            =
 =============================================*/
 
-const cardList = new Section(
-  {
-    items: initialCards,
-    renderer: createCard,
-  },
-  ".cards__list"
+const profileImagePopupForm = new PopupWithForm(
+  "#picture-modal",
+  handleProfilePictureEdit
 );
-cardList.renderItems();
+profileImagePopupForm.setEventListeners();
 
-function handleAddCardEditSubmit(cardData) {
-  const name = cardData.title;
-  const link = cardData.url;
-  cardList.addItem({ name, link });
+const profileImageEditButton = document.querySelector(
+  ".profile-image__edit-button"
+);
+
+profileImageEditButton.addEventListener("click", (evt) => {
+  evt.preventDefault();
+  profileImagePopupForm.open();
   addFormValidator.disableButton();
-  cardPopupForm.close();
-  addCardFormElement.reset();
+});
+
+function handleProfilePictureEdit(userData) {
+  profileImagePopupForm.setLoading(true);
+  api
+    .fetchProfilePicture(userData.profileurl)
+    .then((userData) => {
+      profileImagePopupForm.close();
+      userInfo.setUserImage(userData.avatar);
+    })
+    .catch(console.error)
+    .finally(() => profileImagePopupForm.setLoading(false));
+}
+
+/*=============================================
+=            new card            =
+=============================================*/
+
+const cardPopupForm = new PopupWithForm(
+  "#add-card-modal",
+  handleNewCardFormSubmit
+);
+cardPopupForm.setEventListeners();
+
+function handleNewCardFormSubmit(userInfo) {
+  cardPopupForm.setLoading(true);
+  api
+    .fetchNewCard(userInfo)
+    .then((res) => {
+      cardList.addItem(res);
+    })
+    .then(() => {
+      addCardFormElement.reset();
```

Comment:
`addFormValidator.disableButton();` should be only here

Reply:
on this one if i remove ( cardPopupForm.close(); ) it wont close the modal after i submit the new card

Reply:
and  addCardFormElement.reset(); is what i use to cleat the inputs so if taek both those away it wont close after submit or take away the inputs